src/TorchSharp/Tensor/Tensor.cs

```diff
@@ -2840,6 +2860,14 @@ public Tensor eq_(Scalar target)
 
             public bool Equals(Tensor target)
             {
+                if (handle == IntPtr.Zero || target.handle == IntPtr.Zero) {
+                    return (object)this == (object)target;
+                }
+
+                if (dtype != target.dtype) {
+                    return false;
+                }
+
```

Comment:
As I mentioned above, you should not be hacking `Equals` for your `HashSet`. Instead, use `ReferenceEqualityComparer`.

Reply:
I reverted this change in the latest version